### PR TITLE
Add unit test to not log stdout or stderr fields on create task request

### PR DESCRIPTION
**Pull request by austinvazquez**
*Issue #, if available:*
Validates #781 

*Description of changes:*
This change adds a unit test to validate when launching a container with the firecracker runtime the shim implementation will not log the stdout/stderr shim logger binary URI to disk.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: runtime/service_test.go [hunk 1]
```diff
@@ -323,3 +327,42 @@ func TestDebugConfig(t *testing.T) {
...
 }
+
+// TestTaskCreateDoesNotLogStdoutOrStderrFields validates the firecracker-containerd runtime service implementation
+// does not log the Stdout/Stderr request fields because they may contain sensitive information for the shim logger binary use case.
+func TestTaskCreateDoesNotLogStdoutOrStderrFields(t *testing.T) {
+	testLogger, hook := test.NewNullLogger()
+	testLogger.Level = logrus.DebugLevel
+
+	vmIsReady := make(chan struct{})
+	close(vmIsReady)
```
Comment (sondavidb) [inline runtime/service_test.go:338]: `defer close(vmisReady)`?

**Review by sondavidb — COMMENTED** [review]
Is there a mechanism for not redacting the info? (e.g. if a debug flag is set) If so, we might want to make sure that it gets tested as well.

File: runtime/service_test.go (hunk 1 above)
Comment (austinvazquez) [reply runtime/service_test.go:338]: In this case, the channel is closed after the `CreateVM` API call is made. So mocking here the VM is ready.

**Comment by austinvazquez** [comment]
> Is there a mechanism for not redacting the info? (e.g. if a debug flag is set) If so, we might want to make sure that it gets tested as well.

Not that I have found, but open to ideas there.

File: runtime/service_test.go [hunk 2]
```diff
@@ -323,3 +327,42 @@ func TestDebugConfig(t *testing.T) {
...
+	createTaskRequest := &task.CreateTaskRequest{
+		ID:     t.Name(),
+		Stdout: "/sbin/shim-loggers-for-containerd --env USERNAME=admin --env PASSWORD=admin",
+		Stderr: "/sbin/shim-loggers-for-containerd --env TOKEN=tolkien",
+	}
+
+	// (*service).Create will fail on (Dir).CreateBundleLink after the log we want to validate.
+	_, _ = uut.Create(ctx, createTaskRequest)
+
+	for _, entry := range hook.AllEntries() {
```
[the remark this answers is not in the document]
Comment (austinvazquez) [reply runtime/service_test.go:355]: Good call.

**Review by swagatbora90 — APPROVED** [review]

**Review by sondavidb — APPROVED** [review]

**Pull request merged**